File: loadercomplex-fabric/ver17/src/main/java/com/enderzombi102/loadercomplex/fabric17/impl/entity/FabricItemEntity.java
package com.enderzombi102.loadercomplex.fabric17.impl.entity;

import com.enderzombi102.loadercomplex.fabric17.impl.item.FabricItemStack;
import com.enderzombi102.loadercomplex.minecraft.entity.ItemEntity;
import com.enderzombi102.loadercomplex.minecraft.item.ItemStack;

public class FabricItemEntity extends FabricEntity implements ItemEntity {
	private final net.minecraft.entity.ItemEntity backingEntity;

	public FabricItemEntity(net.minecraft.entity.ItemEntity backingEntity) {
		super( backingEntity );
		this.backingEntity = backingEntity;
	}

	@Override
	public ItemStack getStack() {
		return new FabricItemStack( this.backingEntity.getStack() );
	}

	@Override
	public void setStack(ItemStack stack) {
		this.backingEntity.setStack( (net.minecraft.item.ItemStack) stack.getStack() );
	}
}

File: loadercomplex-fabric/src/ver122/kotlin/com/enderzombi102/loadercomplex/fabric122/impl/entity/FabricItemEntity.kt
package com.enderzombi102.loadercomplex.fabric122.impl.entity

import com.enderzombi102.loadercomplex.api.minecraft.entity.ItemEntity
import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack
import com.enderzombi102.loadercomplex.fabric122.impl.item.FabricItemStack
import com.enderzombi102.loadercomplex.fabric122.impl.utils.toMC
import net.minecraft.entity.ItemEntity as McItemEntity

class FabricItemEntity( private val backingEntity: McItemEntity ) : FabricEntity( backingEntity ), ItemEntity {
	override fun getStack(): ItemStack {
		return FabricItemStack(backingEntity.item)
	}

	override fun setStack( stack: ItemStack ) {
		backingEntity.item = stack.toMC()
	}
}
